adapters/http-vertx/src/main/java/org/eclipse/hono/adapter/http/vertx/VertxBasedHttpProtocolAdapter.java

```diff
@@ -49,16 +51,26 @@ protected String getTypeName() {
 
     @Override
     protected final void addRoutes(final Router router) {
+        setupCorsHandler(router);
+
         if (!getConfig().isAuthenticationRequired()) {
             LOG.warn("device authentication has been disabled");
             LOG.warn("any device may publish data on behalf of all other devices");
         } else {
             setupBasicAuth(router);
         }
+
         addTelemetryApiRoutes(router);
         addEventApiRoutes(router);
     }
 
+    private void setupCorsHandler(final Router router) {
+        router.route()
+                .handler(CorsHandler.create(getConfig().getCorsAllowedOrigin()).allowedMethod(HttpMethod.PUT)
+                        .allowedMethod(HttpMethod.POST).allowedMethod(HttpMethod.OPTIONS)
+                        .allowedHeader("Authorization").allowedHeader("Content-Type"));
```

Comment:
Please use the corresponding constants for specifying the allowed headers:
`io.vertx.core.http.HttpHeaders.AUTHORIZATION` and `io.vertx.core.http.HttpHeaders.CONTENT_TYPE`.
IMHO we do not need to explicitly allow the `Options` method since this is already implicitly allowed (and implemented) by the CorsHandler, isn't it?